src/number_theory/cyclotomic/primitive_roots.lean

```diff
@@ -161,20 +162,44 @@ section norm
 namespace is_primitive_root
 
 variables [field L] {ζ : L} (hζ : is_primitive_root ζ n)
+variables {K} [field K] [algebra K L] [ne_zero ((n : ℕ) : K)]
+
+/-- If `finrank K L = 1`, then the norm of a `2`-th primitive root of unity is `-1`. This
+mathematically trivial result is complementary to `norm_eq_one` below. -/
+lemma norm_eq_neg_one (h : finrank K L = 1) (hζ : is_primitive_root ζ 2) :
+  norm K ζ = -1 :=
+by rw [is_primitive_root.eq_neg_one_of_two_right hζ, show -1 = algebra_map K L (-1), by simp,
+      norm_algebra_map, h, pow_one]
 
 include hζ
 
-/-- If `K` is linearly ordered (in particular for `K = ℚ`), the norm of a primitive root is `1`
-if `n` is odd. -/
-lemma norm_eq_one [linear_ordered_field K] [algebra K L] (hodd : odd (n : ℕ)) : norm K ζ = 1 :=
```

Comment:
these two seem to be disjoint requirements, right? mathematically, can we have a linear ordered field that has a non-irreducible cyclotomic polynomial? if so, I think we can keep both around, perhaps with slightly different names

Reply:
Having it certainly doesn't hurt.